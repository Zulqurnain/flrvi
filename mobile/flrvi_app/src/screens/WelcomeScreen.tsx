import React from 'react';
import { View, Text, StyleSheet, Button, Image } from 'react-native';

const WelcomeScreen = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.logo}>[Logo Placeholder]</Text>
      <Text style={styles.tagline}>Find Your Connection in Thailand.</Text>
      <View style={styles.buttonContainer}>
        <Button
          title="Create Account"
          onPress={() => navigation.navigate('RegisterScreen')}
          color="#E91E63"
        />
      </View>
      <View style={styles.buttonContainer}>
        <Button
          title="Login"
          onPress={() => navigation.navigate('LoginScreen')}
          color="#FF9800"
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
    padding: 24,
  },
  logo: {
    width: 150,
    height: 150,
    marginBottom: 20,
  },
  tagline: {
    fontSize: 20,
    fontWeight: '600',
    color: '#333333',
    textAlign: 'center',
    marginBottom: 30,
  },
  buttonContainer: {
    width: '100%',
    marginVertical: 8,
  },
});

export default WelcomeScreen;
