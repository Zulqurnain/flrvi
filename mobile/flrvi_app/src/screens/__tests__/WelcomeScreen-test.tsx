import React from 'react';
import renderer from 'react-test-renderer';
import WelcomeScreen from '../WelcomeScreen';

it('renders correctly', () => {
  const navigation = { navigate: jest.fn() };
  const tree = renderer.create(<WelcomeScreen navigation={navigation} />).toJSON();
  expect(tree).toMatchSnapshot();
});
