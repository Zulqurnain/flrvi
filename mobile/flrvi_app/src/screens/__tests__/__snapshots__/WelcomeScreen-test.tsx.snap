// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
    }
  }
>
  <Text
    style={
      {
        "height": 150,
        "marginBottom": 20,
        "width": 150,
      }
    }
  >
    [Logo Placeholder]
  </Text>
  <Text
    style={
      {
        "color": "#333333",
        "fontSize": 20,
        "fontWeight": "600",
        "marginBottom": 30,
        "textAlign": "center",
      }
    }
  >
    Find Your Connection in Thailand.
  </Text>
  <View
    style={
      {
        "marginVertical": 8,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              {
                "color": "#E91E63",
              },
            ]
          }
        >
          Create Account
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "marginVertical": 8,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              {
                "color": "#FF9800",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
